packages/validators/ts-typechecker/src/typecheckValidator.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import path from 'path';
+import {resolveConfig} from '@parcel/utils';
+import {Validator} from '@parcel/plugin';
+import localRequire from '@parcel/local-require';
+
+import formatDiagnostics from './formatDiagnostics';
+import LanguageServiceHost from './languageServiceHost';
+
+let langServiceCache = {};
+
+export default new Validator({
+  async validate({asset, options}) {
+    let ts = await localRequire('typescript', asset.filePath);
+
+    let configNames = ['tsconfig.json'];
+    // $FlowFixMe
+    let configPath = await resolveConfig(asset.filePath, configNames);
+
+    if (!langServiceCache[configPath]) {
```

Comment:
Does this cache ever need to be invalidated? i.e. when the config changes? Can be part of the overall config redesign later.

Reply:
If it needs to consider the changes while running it definitely needs a way to invalidate this

Reply:
Added an object hash so this should work now